src/tcp_connections.rs

```diff
@@ -53,7 +53,13 @@ impl <T> Drop for OutTcpStream<T> {
 /// Connect to a peer and open a send-receive pair.  See `upgrade` for more details.
 pub fn connect_tcp<'a, 'b, I, O>(addr: SocketAddr) -> IoResult<(Receiver<I>, OutTcpStream<O>)>
         where I: Send + Decodable + 'static, O: Encodable {
-    Ok(try!(upgrade_tcp(try!(TcpStream::connect(&addr)))))
+    let stream = try!(TcpStream::connect(&addr));
+    if try!(stream.peer_addr()).port() == try!(stream.local_addr()).port() {
```

Comment:
With this we restrict same src/dst port connection. As crust doesn't provide src port in it's API, this should be okay.

We need to document the behaviour in API so that upper layer retries if it gets ConnectionRefused error.
Other option is to retry connect internally and it will likely pick a different port.

Will be good to hear from @dirvine